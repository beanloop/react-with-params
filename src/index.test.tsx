import {render} from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import Router from 'react-router/MemoryRouter'
import {withParams} from './index'

describe('withParams', () => {
  it('should extract a single param', () => {
    const ShowName = withParams({params: 'name', match: '/user/:name'})(({name}) =>
      <span>{name}</span>
    )

    const wrapper = render(
      <Router initialEntries={['/user/ben']}>
        <ShowName />
      </Router>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should extract multiple params', () => {
    const ShowIdAndName = withParams({params: ['id', 'name'], match: '/user/:id/:name'})(({id, name}) =>
      <span>{id} - {name}</span>
    )

    const wrapper = render(
      <Router initialEntries={['/user/1/ben']}>
        <ShowIdAndName />
      </Router>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should extract params with exactly false', () => {
    const ShowIdAndName = withParams({
      params: ['id', 'name'],
      match: '/user/:id/:name',
      exactly: false,
    })(({id, name}) =>
      <span>{id} - {name}</span>
    )

    const wrapper = render(
      <Router initialEntries={['/user/1/ben/the/third']}>
        <ShowIdAndName />
      </Router>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should extract a single query parameter', () => {
    const Display = withParams({
      queryParams: 'next',
      match: '/users',
    })(({next}) =>
      <span>{next}</span>
    )

    const wrapper = render(
      <Router initialEntries={['/users?next=/login']}>
        <Display />
      </Router>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should extract multiple query parameters', () => {
    const Display = withParams({
      queryParams: ['next', 'timestamp'],
      match: '/users',
    })(({next, timestamp}) =>
      <span>{next} - {timestamp}</span>
    )

    const wrapper = render(
      <Router initialEntries={['/users?next=/login&timestamp=1489995623368']}>
        <Display />
      </Router>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
