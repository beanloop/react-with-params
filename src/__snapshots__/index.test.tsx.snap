// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withParams should extract a single param 1`] = `
<span>
  ben
</span>
`;

exports[`withParams should extract a single query parameter 1`] = `
<span>
  /login
</span>
`;

exports[`withParams should extract multiple params 1`] = `
<span>
  1 - ben
</span>
`;

exports[`withParams should extract multiple query parameters 1`] = `
<span>
  /login - 1489995623368
</span>
`;

exports[`withParams should extract params with exactly false 1`] = `
<span>
  1 - ben
</span>
`;
